kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/security/CertificateListTabUi.java

```diff
@@ -234,111 +237,63 @@ public String getValue(GwtKeystoreEntry object) {
         this.certificatesGrid.setSelectionModel(this.selectionModel);
     }
 
+    private <U extends Comparable<U>> Comparator<GwtKeystoreEntry> getComparator(Function<GwtKeystoreEntry, U> comparableElementSupplier) {
+    	return new Comparator<GwtKeystoreEntry>() {
+
+			@Override
+			public int compare(GwtKeystoreEntry o1, GwtKeystoreEntry o2) {
+				if(o1 == o2)
+					return 0;
+				if(o1 == null)
+					return -1;
+				if(o2 == null)
+					return 1;
+				
+				U item1 = comparableElementSupplier.apply(o1);
+				U item2 = comparableElementSupplier.apply(o2);
+				
+				return item1.compareTo(item2);
```

Comment:
Probably we should also handle the case when `item1` or `item2` are `null`, (e.g `getValidityStartDate` returns `null`)

Reply:
right, I missed that case